tests/test_marathon.py

```diff
@@ -21,11 +21,6 @@ def test_add_pod_returns_parsed_response_body():
     _assert_add_pod_returns_parsed_response_body(["another", "pod", "json"])
 
 
-def test_add_pod_raises_dcos_exception_for_json_parse_errors():
-    _assert_method_raises_dcos_exception_for_json_parse_errors(
-        lambda marathon_client: marathon_client.add_pod({'some': 'json'}))
```

Comment:
Since you're now returning the header, let's add a test to cover the case when the header value is missing (we want to make sure the user doesn't see a Python stacktrace when things fail). See the `update_pod()` method in `dcos/marathon.py` for code that handles it already (you might want to factor that out into its own method), and the corresponding unit test in this file: `test_update_pod_raises_dcos_exception_if_deployment_id_missing()`.